### PR TITLE
Documentation

### LINKED ISSUE #92: Documentation
Set up (semi-)automatic creation of documentation

### PR DESCRIPTION
**Pull request by schote**
- Setup of documentation pipeline to automatically deploy documentation to github-pages
- Setup pydata theme
- Implementation of templates for modules and classes: Using autosummary the api documentation is generated automatically using the templates
- Basic page structure
  - documentation
  - user guide
  - contributer guide
  - api

closes #92 

Documentation page: https://ptb-mr.github.io/mrpro/

### CODE REVIEW

File: docs/source/_static/img/logo.jpg
Comment (fzimmermann89) [inline docs/source/_static/img/logo.jpg:1]: Is there maybe a smaller (filesize) non-jpeg  version of the ptb logo? preferebly a vector graphic?

Comment (fzimmermann89) [reply docs/source/_static/img/logo.jpg:1]: ptb_logo
(if we are even allowed to use the ptb logo without the PTB letters)

Comment (fzimmermann89) [reply docs/source/_static/img/logo.jpg:1]: are we allowed to use the logo without the text?

**Comment by fzimmermann89** [comment]
ptb_logo

File: docs/source/_static/img/logo.jpg
Comment (schote) [reply docs/source/_static/img/logo.jpg:1]: I will have a look tomorrow 👍🏼

**Review by schuenke — APPROVED** [review]
In general the (semi-automatic) documentation seems to work well. I discussed and tried several things with @fzimmermann89 yesterday and these are a few points we might want to adjust/change in the future:

- Add the Parameters/Attributes of all (data)classes to the docstring to have them documented in a nicer why in the html docu and, even more important, provide useful tool tips in the IDE when creating instances. For example:
```python
@dataclass(slots=True)
class IHeader:
    """MR image data header.

    Parameters
    ----------
    fov
        Field of view in [m].
    te
        Echo time in [s].
    ti
        Inversion time in [s].
    fa
        Flip angle in [rad].
    """
```

- remove `.. automethod:: __init__` from the templates
- separate classmethods and methods in the templates
- discuss the general theme (e.g., pydata, read the docs etc.)

However, this should all be adjustable later on and is thus low-priority imo. I am fine with merging it as it is for the moment.

**Comment by ckolbPTB** [comment]
Please do not use the PTB logo!

**Comment by ckolbPTB** [comment]
If I go to https://ptb-mr.github.io/mrpro/ then all I see is a nicely formatted readme page (see below). How do I get to the documentation?

Bildschirmfoto 2023-10-18 um 08 35 44

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -3,10 +3,10 @@ name: Sphinx Docs
 on:
   push:
     branches:
-      - documentation
+      - main
   pull_request:
     branches:
-      - documentation
+      - main
```
Comment (fzimmermann89) [inline .github/workflows/docs.yaml]: this is imho wrong.
please double check @schuenke 

will this run on all pull request made against main and overwrite the public documenation?

it should not run the publish to github pages on PR!

**Review by fzimmermann89 — CHANGES_REQUESTED** [review]

File: docs/source/_static/img/logo.jpg
Comment (fzimmermann89) [reply docs/source/_static/img/logo.jpg:1]: seems like we are not supposed to :D

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+sys.path.insert(0, os.path.abspath('../../src'))  # Source code dir relative to this file
+
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'MRPro'
+copyright = '2023, Physikalisch-Technische Bundesanstalt (PTB) Berlin'
+author = 'Christoph Kolbitsch, Patrick Schuenke, Felix Zimmermann, David Schote'
+release = '0.0.1'  # TODO: import this from package
```
Comment (fzimmermann89) [inline docs/source/conf.py]: https://github.com/sphinx-toolbox/sphinx-pyproject

**Comment by fzimmermann89** [comment]
> If I go to https://ptb-mr.github.io/mrpro/ then all I see is a nicely formatted readme page (see below). How do I get to the documentation?

fixed. was a settings issue:
you set the pages to be buld from 'main', not from the github-pages branch.

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (schote) [reply .github/workflows/docs.yaml]: Yes, it runs on pushes and pull requests against main. I though this is useful to see the final result (at least for this PR). Removed the pull_request dependency.

File: docs/source/conf.py (hunk 2 above)
Comment (schote) [reply docs/source/conf.py]: Added to the project!

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (schote) [reply .github/workflows/docs.yaml]: Please notice, that the documentation pipeline is not triggered by this PR anymore due to this change request. Pipeline should be triggered once we completed the merge.

**Review by fzimmermann89 — APPROVED** [review]

**Pull request merged**